### PR TITLE
[4.1] redirect links index

### PR DESCRIPTION
**Pull request by brianteeman**
Set the correct index name

@richard67 if accepted would we need to do an update sql as well?

### CODE REVIEW

**Comment by richard67** [comment]
@brianteeman Yes. And it needs 2 separate statements, one to drop with index with tie old name and another one to create the index with the new name, because the database checker can't handle index renaming yet.

**Comment by richard67** [comment]
If the same index had been created in an old 4.x update SQL it would even need to change that, too, but that's not the case here, I think.

**Comment by brianteeman** [comment]
Should I do that here or in a separate pr ?

**Comment by richard67** [comment]
Better here. If necessary I will suggest changes then.

**Comment by richard67** [comment]
Update SQL scripts look good to me.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 38488bf079d7d992cb1f8c7adfd4eb883d5ac256<br /><br />Testing instructions for other testers: Update a clean, current 4.1 (latest nightly or current 4.1-dev) to the custom update URL or the update package created by drone for this PR. After the update, check that the index has the right name and that the database checker shown only the usual error about the update version not matching the CMS version, which is normal for the patched packages built buy drone, but no other error.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36817">issues.joomla.org/tracker/joomla-cms/36817</a>.</sub>

**Comment by richard67** [comment]
@brianteeman Could you provide some testing instructions? You can find inspiration in my previous comment with my test result. Thanks in advance.

**Comment by richard67** [comment]
P.S. I've tested with both MySQL and PostgreSQL.

File: administrator/components/com_admin/sql/updates/postgresql/4.1.0-2022-01-24.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DROP INDEX IF EXISTS "#__redirect_links_idx_link_modifed";
+CREATE INDEX "#__redirect_links_idx_link_modified" on "#__redirect_links" ("modified_date");
```
Comment (Quy) [inline administrator/components/com_admin/sql/updates/postgresql/4.1.0-2022-01-24.sql]:
```suggestion
CREATE INDEX "#__redirect_links_idx_link_modified" ON "#__redirect_links" ("modified_date");
```

Comment (richard67) [reply administrator/components/com_admin/sql/updates/postgresql/4.1.0-2022-01-24.sql]: Agree, not a problem but code style.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 86fc6412ed06a5ff64d7163b17be02a5fd63ddd2<br /><br />My previous test is still valid since the change after that was just SQL code style.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36817">issues.joomla.org/tracker/joomla-cms/36817</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 86fc6412ed06a5ff64d7163b17be02a5fd63ddd2<br /><br />Tested on MySQL<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36817">issues.joomla.org/tracker/joomla-cms/36817</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36817">issues.joomla.org/tracker/joomla-cms/36817</a>.</sub>

**Comment by pritam825** [comment]
sir, the same code  is in 4.0 so, can I open the pr for this one also?

**Comment by bembelimen** [comment]
Thx

**Comment by brianteeman** [comment]
thanks

**Pull request merged**